### ISSUE TITLE
I hope that MLT can support concurrent based on playlist?

### ISSUE THREAD

**Issue by davidwongiiss1984** [mltframework/mlt#446]
I hope that mlt can support concurrent(pipeline) based on playlist, like below...
whether it is feasible? and Is there a better way? Please enlighten me!  MLT is amazing!!! 👍 
thank you very much!

1. create main and sub consumers (hook get_frame of playlist with new_get_frame)
2. start and running consumer (real_time=1) ...

``` c++
main consumer -> tractor
                   transition
                   multitrack
  sub_consumer1 -> playlist1
                     new_get_frame
                       if (running in ahead_thread of main) 
                         mlt_consumer_rt_frame(sub_consumer1)
                       else 
                         playlist->get_frame 
  sub_consumer2 -> playlist2
                     blablabla...

**Comment by davidwongiiss1984** [comment]
My engine based on movit, "consumer-thread-create" listener pass engine as "void* service"
```C++
m_consumer->listen("consumer-thread-create", engine, (mlt_listener)onThreadCreate);
```
I need create offscreen in current read_ahead thread, this is why I need to know which consumer the "consumer-thread-started" event comes from. #443

**Comment by ddennedy** [comment]
This will not work. There can only be one consumer per service graph at the same time. You can have multiple consumers and switch between them by stop, disconnect, connect, start. You can also have multiple graphs with their own consumer in the same process **except** in the case of opengl/Movit. With Movit, there is a global instance of `GlslManager` from src/modules/opengl/filter_glsl_manager.cpp.
In that case, I recommend you figure out some multi-process model for your app.

**Comment by davidwongiiss1984** [comment]
Thank you very much!

**Comment by davidwongiiss1984** [comment]
Yes, my goal is to reduce the time spent on each frame during real-time editing.
I have implemented multiple instances of GlslManager and instance management based on thread context. Each playlist creates a separate sub-consumer and movit chain.
My idea is to generate the texture of each playlist in parallel and then connect them on the main consumer.
Now I have successfully exported the texture of each playlist concurrently based on the above instructions.
Next I plan to connect them...
